### ISSUE TITLE
Distance to "no data"?

### ISSUE THREAD

**Issue by stenvall** [DotSpatial/DotSpatial#915]
Hello,
I am trying to create a polygon that represents all area within the extent of a FeatureSet that does not have a polygon. That is a polygon that represents missing data within a certain extent. The purpose is to include missing data as a feasible target location when calculating certain distances.
I have managed to do so but when I have a large FeatureSet this operation takes a lot of time. 
Any suggestions besides parallelization?

        private IGeometry getNoDataPolygon(IFeatureSet featureSet)
        {

                IGeometry NoDataPolygon = featureSet.Extent.ToEnvelope().ToPolygon();
                
                foreach (IFeature feature in featureSet.Features)
                {
                    NoDataPolygon = NoDataPolygon.Difference(Geometry.FromBasicGeometry(feature.BasicGeometry));
                }

                return NoDataPolygon;

        }

**Comment by mogikanin** [comment]
Yes, you can use `Parallel.ForEach` instead of your foreach loop. It will give some speed improvement on multicore machines.

**Comment by mogikanin** [comment]
Ah, I missed "besides" :)

**Comment by stenvall** [comment]
I tried using EraseFeatures in Overlay

        private IGeometry getNoDataPolygon(IFeatureSet featureSet)
        {

                featureSet.InitializeVertices();
                FeatureSet noDataFeatureSet = new FeatureSet();
                IGeometry NoDataPolygon = featureSet.Extent.ToEnvelope().ToPolygon();
                noDataFeatureSet.AddFeature(NoDataPolygon);
                noDataFeatureSet.InitializeVertices();
                noDataFeatureSet = Overlay.EraseFeatures(noDataFeatureSet, featureSet);
                return Geometry.FromBasicGeometry(noDataFeatureSet.Features[0].BasicGeometry);
        }


but this was just as slow.  The EraseFeatures method is using the Differance metod i used in previous example so I should have guessed. But instead of looping the features directly it uses the shape indices as a memory management technique.

**Comment by jany-tenaj** [comment]
@stenvall Have you tried to union your existing polygons before calling difference? I think Union + a single Difference call might be faster.

It might be usefull to switch to DS 2.0 to get rid of `Geometry.FromBasicGeometry(feature.BasicGeometry)`.
In DS 2.0 you can use feature.Geometry directly and don't have to recreate it.

**Comment by stenvall** [comment]
@jany-tenaj Thanks for the tip! I tried to union the existing polygons + a single Difference call but it was allso werry slow. It is not suitable for us to switch to DS 2.0 right now but maybe that is the only way forward. Any other suggestions?

**Comment by jany-tenaj** [comment]
@stenvall just found [another answer](https://github.com/NetTopologySuite/NetTopologySuite/issues/139) for you. But I fear you'd have to switch to DS 2.0 for this, too.

**Comment by mogikanin** [comment]
I guess question is solved, and we can close it?
